src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9278,6 +9278,13 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                 receiver = ReplaceTypeOrValueReceiver(typeOrValue, useType, diagnostics);
             }
 
+            if (receiver.Type.IsRefLikeType)
+            {
+                // Cannot perform a dynamic call on a ref struct '{0}'.
+                Error(diagnostics, ErrorCode.ERR_CannotDynamicInvokeOnRefStruct, receiver.Syntax, receiver.Type);
```

Comment:
>ERR_CannotDynamicInvokeOnRefStruct [](http://example.com/codeflow?start=45&length=34)

We probably could reuse `ERR_BadDynamicMethodArg` adjusting its wording to mention a receiver as well.

Reply:
I'd prefer the more precise error for this case.